errors/errors.go

```diff
@@ -66,6 +66,27 @@ var (
 	ErrInvalidOutOfBounds = errors.New("data is out of schema bounds")
 )
 
+// Translation errors. These all occur as a result of an invalid lens. Currently
+// these may be returned from [thema.Instance.TranslateErr]. Eventually, it is
+// hoped that they will be caught statically in [thema.BindLineage] and cannot
+// occur at runtime.
```

Comment:
Is that aspiration fair as soon as we foresee the need of adding support for "delegated" translations? Or would that be another, separated, API? 🤔 

NB: By "delegated" translations, I mean a translation that cannot be performed just with plain CUE but relies on external systems (like a DB) and/or requires another way to define it (in Go, for instance).

Reply:
> which would be lovely from a "keep thema's Go interface stable" perspective.

Perhaps adding functional options? I'd say it's quite idiomatic (fair?) and would leave us more flexibility.

> making that degraded mode a normal part of the contract means that a non-erroring Translate() is just permanently off the table

Right, yeah! 😅
But I'm curious, how do you think that'd change with lacunas? 
Would it be something like: errors vs "unsatisfied/unhandled" lacunas?